payments-core/src/main/java/com/stripe/android/model/parsers/ConsumerPaymentDetailsJsonParser.kt

```diff
@@ -0,0 +1,51 @@
+package com.stripe.android.model.parsers
+
+import com.stripe.android.core.model.StripeJsonUtils.optString
+import com.stripe.android.core.model.parsers.ModelJsonParser
+import com.stripe.android.model.CardBrand
+import com.stripe.android.model.ConsumerPaymentDetails
+import org.json.JSONObject
+
+internal class ConsumerPaymentDetailsJsonParser : ModelJsonParser<ConsumerPaymentDetails> {
+    override fun parse(json: JSONObject): ConsumerPaymentDetails {
+        val paymentDetails = json.optJSONArray(FIELD_PAYMENT_DETAILS)
+            ?.let { paymentDetailsArray ->
+                (0 until paymentDetailsArray.length())
+                    .map { index -> paymentDetailsArray.getJSONObject(index) }
+                    .mapNotNull { parsePaymentDetails(it) }
+            } ?: emptyList()
+        return ConsumerPaymentDetails(paymentDetails)
+    }
+
+    private fun parsePaymentDetails(json: JSONObject): ConsumerPaymentDetails.PaymentDetails? =
+        optString(json, FIELD_TYPE)?.let {
```

Comment:
Might have missed this at some point but does link only support cards?

Reply:
Currently yes, it will support bank debit soon and then this will be updated.